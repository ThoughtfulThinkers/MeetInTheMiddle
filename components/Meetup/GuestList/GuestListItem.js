import React, { Component } from 'react';
import { Text } from 'react-native';
import { CardSection } from '../../common';

class GuestListItem extends Component {
  render() {
    return (
      <CardSection style={{ justifyContent: 'center' }}>
        <Text style={styles.titleStyle}>
          {this.props.guest}
        </Text>
      </CardSection>
    );
  }
}

const styles = {
  titleStyle: {
    fontSize: 18
  }
};

export default GuestListItem;
