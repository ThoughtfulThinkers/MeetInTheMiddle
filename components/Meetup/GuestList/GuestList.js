import React, { Component } from 'react';
import { ListView } from 'react-native';
import GuestListItem from './GuestListItem';

class GuestList extends Component {
  componentWillMount() {
    this.createDataSource(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.createDataSource(nextProps);
  }

  createDataSource({ guests }) {
    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2
    });

    this.dataSource = ds.cloneWithRows(guests);
  }

  renderRow(guest) {
      return <GuestListItem guest={guest} />;
  }

  render() {
    return (
      <ListView
        enableEmptySections
        dataSource={this.dataSource}
        renderRow={this.renderRow}
      />
    );
  }
}

export default GuestList;
