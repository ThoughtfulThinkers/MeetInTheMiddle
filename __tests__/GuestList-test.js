import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import GuestList from '../components/Meetup/GuestList/GuestList';

test('renders GuestList correctly', () => {
  const tree = renderer.create(
    <GuestList guests={[{ name: 'Test User' }, { name: 'Test Two' }]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
