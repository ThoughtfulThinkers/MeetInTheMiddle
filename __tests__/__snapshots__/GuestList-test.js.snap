exports[`test renders GuestList correctly 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  enableEmptySections={true}
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }>
      Test User
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }>
      Test Two
    </Text>
  </View>
</ScrollView>
`;
